packages/flutter/lib/src/material/chip.dart

```diff
@@ -10,154 +10,1373 @@ import 'package:flutter/painting.dart';
+  /// }
+  /// ```
+  ValueChanged<bool> get onSelected;
+
+  /// Color to be used for the chip's background, indicating that it is
+  /// selected.
+  ///
+  /// The chip is selected when [selected] is true.
+  Color get selectedColor;
+
+  /// Tooltip string to be used for the body area (where the label and avatar
+  /// are) of the chip.
+  String get tooltip;
+}
+
+/// An interface for material design chips that can be enabled and disabled.
+///
+/// See also:
+///
+///  * [InputChip], a chip represents a complex piece of information, such as an
+///    entity (person, place, or thing) or conversational text, in a compact
+///    form.
+///  * [ChoiceChip], allows a single selection from a set of options. Choice
+///    chips contain related descriptive text or categories.
+///  * [FilterChip], uses tags or descriptive words as a way to filter content.
+///  * <https://material.google.com/components/chips.html>
+abstract class DisabledChipAttributes {
+  /// Whether or not this chip is enabled for input.
+  ///
+  /// If this is true, but all of the user action callbacks are null (i.e.
+  /// [SelectableChipAttributes.onSelected], [TappableChipAttributes.onPressed],
+  /// and [DeletableChipAttributes.onDelete]), then the
+  /// control will still be shown as disabled.
+  ///
+  /// This is typically used if you want the chip to be disabled, but also show
+  /// a delete button.
+  ///
+  /// For classes which don't have this as a constructor argument, [isEnabled]
+  /// returns true if their user action callback is set.
+  ///
+  /// Defaults to true. Cannot be null.
+  bool get isEnabled;
+
+  /// Color to be used for the chip's background indicating that it is disabled.
+  ///
+  /// The chip is disabled when [isEnabled] is false, or all three of
+  /// [SelectableChipAttributes.onSelected], [TappableChipAttributes.onPressed],
+  /// and [DeletableChipAttributes.onDelete] are null.
+  ///
+  /// It defaults to [Colors.black38].
+  Color get disabledColor;
+}
+
+/// An interface for material design chips that can be tapped.
+///
+/// See also:
+///
+///  * [InputChip], a chip represents a complex piece of information, such as an
+///    entity (person, place, or thing) or conversational text, in a compact
+///    form.
+///  * [ChoiceChip], allows a single selection from a set of options. Choice
+///    chips contain related descriptive text or categories.
+///  * [FilterChip], uses tags or descriptive words as a way to filter content.
+///  * [ActionChip], represents an action related to primary content.
+///  * <https://material.google.com/components/chips.html>
+abstract class TappableChipAttributes {
+  /// Called when the user taps the chip.
+  ///
+  /// If [onPressed] is set, then this callback will be called when the user
+  /// taps on the label or avatar parts of the chip. If [onPressed] is null,
+  /// then the chip will be disabled.
+  ///
+  /// The [onPressed] and [SelectableChipAttributes.onSelected] callbacks must
```

Comment:
This point probably belongs in the implementing class

Reply:
OK, I moved it to the constructors for InputChip and RawChip (the only ones where it applies).